bugsnag/configuration.py

```diff
@@ -8,6 +8,7 @@
 import logging
 from threading import Lock
 
+from os import PathLike
```

Comment:
This was [added in Python 3.6](https://docs.python.org/3/library/os.html#os.PathLike) so the import fails in Python 3.5, which we still support

I think the best we can do on 3.5 is to accept `Union[str, PurePath]` instead, since all the builtin `Path` objects inherit from `PurePath`

We do things like that elsewhere by catching the `ImportError`, e.g.:

https://github.com/bugsnag/bugsnag-python/blob/cc6aa2027b3c152b35b984638f9c749f25e7d39f/bugsnag/sessiontracker.py#L8-L14

Reply:
I've switched it to use PurePath instead of PathLike.